prov/efa/src/efa_fabric.c

```diff
@@ -1023,8 +1023,12 @@ static int efa_init_info(const struct fi_info **all_infos)
 		return ret;
 
 	ctx_list = efa_device_get_context_list(&num_devices);
-	if (!num_devices)
+	if (!num_devices) {
+		if (ctx_list) {
```

Comment:
no need for this `{` and `}' for single line of code.

Reply:
I know it's unnecessary but I've had problems in the past where not encapsulating in a bracket caused bugs (ie. adding more lines into the if statement without adding brackets...won't have a compile error but will be a nasty bug). I like doing this for future proofing.

Reply:
Sure. make sense.